src/lib/bootloader/sections.rb

```diff
@@ -44,7 +42,31 @@ def default=(value)
     # writes default to system making it persistent
     def write
       return if default.empty?
-      Yast::Execute.on_target("/usr/sbin/grub2-set-default", default)
+
+      Yast::Execute.on_target("/usr/sbin/grub2-set-default", title_to_path(default))
+    end
+
+  private
+
+    def read_default
+      # Execute.on_target can return nil if call failed. It shows users error popup, but bootloader
```

Comment:
I mean that Execute.on_target show error popup and return nil. And in such case it use default `""` which is later properly handled as use first available entry.